src/modules/asab-config/ConfigEditor.js

```diff
@@ -0,0 +1,370 @@
+import React, { useState, useEffect } from "react";
+import { useForm } from "react-hook-form";
+import TreeMenu from 'react-simple-tree-menu';
+
+import {
+	Container,
+	Col, Row,
+	Button,
+	ButtonDropdown, DropdownToggle, DropdownMenu, DropdownItem,
+	Card, CardBody, CardHeader, CardFooter, CardTitle, CardSubtitle,
+	Collapse,
+	Form, FormGroup, FormText, Input, Label
+} from "reactstrap";
+
+import {
+	ConfigItem,
+	NumberConfigItem,
+	UrlConfigItem,
+	EmailConfigItem,
+	PasswordConfigItem,
+	CheckBoxConfigItem,
+	RadioButtonConfigItem,
+	SelectConfigItem,
+	TextAreaConfigItem,
+	ConfigAdHocItem,
+} from './ConfigFormatItems';
+
+
+export function ConfigEditor(props) {
+
+	const [ typeData, setTypeData ] = useState(undefined);
+	const [ adHocValues, setAdHocValues ] = useState({});
+	const [ adHocSections, setAdHocSections ] = useState({});
+	const { register, handleSubmit, setValue, getValues, errors, reset } = useForm();
+
+	let App = props.app;
+
+	let urlParams = new URLSearchParams(App.props.location.search);
+	const urlConfigType = urlParams.get("configType");
+	const urlConfigName = urlParams.get("configName");
+
+	const configType = props.configType ? props.configType : urlConfigType;
+	const configName = props.configName ? props.configName : urlConfigName;
+
+	// Retrieve the ASAB_CONFIG_URL from config file
+	let services = App.Config.get('SERVICES');
+	let url = services?.asabconfig ? services.asabconfig : 'asab-config';
+	const Axios = App.axiosCreate(url);
+
+
+	// Load Schema type - it is triggered on every configName change
+	useEffect(() => {
+		if (configType && configName) {
```

Comment:
No if ... please, config_type and config_name HAS TO be present ... if you encounter situations, when they are not, then you have incorrect design.